Sources/AblyChat/Presence.swift

```diff
@@ -1,18 +1,86 @@
 import Ably
 
-// TODO: (https://github.com/ably-labs/ably-chat-swift/issues/13): try to improve this type
-public typealias PresenceData = any Sendable
+public enum PresenceCustomData: Sendable, Codable, Equatable {
```

Comment:
Looks complicated, why it's not just `[String: Any]`? Leads to a lot of noise in `asQueryItems()`.

Reply:
`Any` becomes difficult to encode/decode from too - I'd rather we were particular about what data types are allowed but the spec is very vague here. JS (naturally) allows for less type safety here too. 

I think this is a fine compromise for now. I've also re-added the TODO to revisit this